src/store/matrix/saga.ts

```diff
@@ -72,17 +80,35 @@ export function* generateBackup() {
   }
 }
 
-export function* saveBackup() {
+export function* proceedToVerifyKey() {
+  const existingKey = yield select((state) => state.matrix.generatedRecoveryKey);
+
+  if (existingKey) {
+    yield put(setBackupStage(BackupStage.VerifyKeyPhrase));
+  } else {
+    yield put(setBackupStage(BackupStage.RestoreBackup));
+  }
+}
```

Comment:
This seems like we've mixed two possible saga paths. They should just be separate.

Reply:
Yeah another picky one as they both 'verify' in a sense, but agree, could be improved